sstable/src/dictionary.rs

```diff
@@ -56,6 +56,51 @@ impl Dictionary<VoidSSTable> {
     }
 }
 
+fn map_bound<TFrom, TTo>(bound: &Bound<TFrom>, transform: impl Fn(&TFrom) -> TTo) -> Bound<TTo> {
+    use self::Bound::*;
+    match bound {
+        Excluded(ref from_val) => Bound::Excluded(transform(from_val)),
+        Included(ref from_val) => Bound::Included(transform(from_val)),
+        Unbounded => Unbounded,
+    }
+}
+
+fn map_bound_res<TFrom, TTo>(
+    bound: &Bound<TFrom>,
+    transform: impl Fn(&TFrom) -> io::Result<Bound<TTo>>,
+) -> io::Result<Bound<TTo>> {
```

Comment:
i honestly don't like this function. The spec is odd. It takes a Bound, but then ignores the variant, and gives no way to the transform to know if the bound was inclusive or not. The places this function is used end up with a closure that captures the `Bound<TFrom>` to properly handle Included vs Excluded.

I'm not sure how to make something that's as flexible, while being clearer.
At the very least there should be a doc-comment warning that its behaviour might be surprising.

Reply:
I renamed it to `transform_bound_inner` and added a comment